internal/sync/sync.go

```diff
@@ -0,0 +1,332 @@
+
+		return true
+	}
+
+	return false
+}
+
+func syncDirectories(clonedDir, localDir string, repo *repo) {
+	glog.V(100).Infof("Repos are not synced. Cleaning local directory: %s", localDir)
+
+	if os.RemoveAll(localDir) != nil {
+		glog.V(100).Infof("Failed to remove local api directory. Exist with error code 1")
+		os.Exit(1)
+	}
+
+	glog.V(100).Infof("Create new local directory: %s", localDir)
+
+	if os.MkdirAll(localDir, 0750) != nil {
+		glog.V(100).Infof("Failed to recreate api directory. Exist with error code 1")
+		os.Exit(1)
+	}
+
+	glog.V(100).Infof("Copy api filed from cloned directory to local api directory")
+
+	err := execCmd(
+		"",
+		"cp",
+		[]string{"-a", fmt.Sprintf("%s/.", clonedDir), fmt.Sprintf("%s/", localDir)})
+
+	if err != nil {
+		glog.Infof("failed to sync directories. exist with error code 1")
+		os.Exit(1)
+	}
+
+	glog.V(100).Infof("Fix packages names")
+
+	err = refactor(
+		fmt.Sprintf("package %s", path.Base(clonedDir)),
+		fmt.Sprintf("package %s", path.Base(localDir)),
+		localDir,
+		"*.go")
+
+	if err != nil {
+		glog.V(100).Infof("Failed to refactor file. Exist with error 1")
+		os.Exit(1)
+	}
+
+	for _, importMap := range repo.ReplaceImports {
+		err = refactor(fmt.Sprintf("%q", importMap["old"]), importMap["new"], localDir, "*.go")
+
+		if err != nil {
+			glog.V(100).Infof("Failed to refactor file. Exist with error 1")
+			os.Exit(1)
+		}
+	}
+}
+
+func gitReset(packageName string) {
+	for _, cmdToRun := range [][]string{
+		{"reset", "--", fmt.Sprintf("./pkg/%s", packageName)},
+		{"checkout", "--", fmt.Sprintf("./pkg/%s", packageName)},
+		{"clean", "-d", "-f", fmt.Sprintf("./pkg/%s", packageName)},
+	} {
+		err := execCmd("", "git", cmdToRun)
+
+		if err != nil {
+			glog.Infof("Failed to reset project to it's original state. Exist with error 1")
+			os.Exit(1)
+		}
+	}
+}
+
+func gitClone(localPath string, repo *repo) {
+	glog.V(100).Infof("Cloning repo %s from %s", repo.Name, repo.RepoLink)
+	localDirectory := path.Join(localPath, repo.Name)
+
+	if _, err := os.Stat(localDirectory); !os.IsNotExist(err) {
+		glog.V(100).Infof(
+			"Local directory already exist for the project: %s. Removing directory", repo.Name)
+
+		err := os.RemoveAll(localDirectory)
+
+		if err != nil {
+			glog.V(100).Infof("Failed to remove repo directory %s due to %w exit with error code 1",
+				localDirectory, err)
+			os.Exit(1)
+		}
+	}
+
+	for idx, gitArgs := range [][]string{
+		{"clone", "-n", "--depth=1", "--filter=tree:0", "-b", repo.Branch, repo.RepoLink},
+		{"sparse-checkout", "set", "--no-cone", repo.RemoteAPIDirectory}, {"checkout"}} {
+		directory := localDirectory
+
+		// first cmd is git clone, it is necessary to run this cmd in parents directory in order to create repo's dir.
+		if idx == 0 {
+			directory = localPath
+		}
+
+		err := execCmd(directory, "git", gitArgs)
+
+		if err != nil {
+			glog.V(100).Infof("Failed to clone repo due to cmd error. Exit with error code 1")
+			os.Exit(1)
+		}
```

Comment:
I don't think we really gain anything by using the args in a loop, it just makes the code harder to follow. I think it would be simpler if we just ran the clone and checkout outside of a loop